### PR TITLE
Added RBAC for releases and packages signing

### PR DESCRIPTION
**Pull request by Korulag**

### CODE REVIEW

File: alws/models.py [hunk 1]
```diff
@@ -226,6 +226,24 @@ def validate_permissions(permissions: int):
 )
 
 
+PlatformRoleMapping = sqlalchemy.Table(
+    'platform_role_mapping',
+    Base.metadata,
+    sqlalchemy.Column(
+        'platform_id',
+        sqlalchemy.Integer,
+        sqlalchemy.ForeignKey('platforms.id'),
```
Comment (anfimovdm) [inline alws/models.py:235]: please, add names for foreign keys

File: scripts/manage_team_roles_for_user.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                )).scalars().all()
+
+            if add_roles:
+                user.roles.extend(add_roles)
+            if remove_roles:
+                roles_ids = [r.id for r in remove_roles]
+                await db.execute(delete(models.UserRoleMapping).where(
+                    models.UserRoleMapping.c.role_id.in_(roles_ids)))
+
+            db.add(user)
```
Comment (anfimovdm) [inline scripts/manage_team_roles_for_user.py:78]: does it work without `db.commit()`?

**Review by anfimovdm — CHANGES_REQUESTED** [review]

File: alws/app.py [hunk 3]
```diff
@@ -31,6 +33,15 @@
 graceful_terminate_event = threading.Event()
 
 
+@app.middleware('http')
+async def permissions_middleware(request: Request, call_next):
```
Comment (javihernandez) [inline alws/app.py:37]: Where is this being used? I can't find any reference to it in the changes

File: alws/crud/build.py [hunk 4]
```diff
@@ -40,7 +41,7 @@ async def create_build(
     if not user:
         raise ValueError(f'Cannot find user with id {user_id}')
 
-    if not can_perform(product, user, 'create_build'):
+    if not can_perform(product, user, actions.CreateBuild.name):
```
Comment (javihernandez) [inline alws/crud/build.py:44]: Not critical, and I do know that `can_perform` is already in the codebase, but to me, I'd prefer to read `can_perform(action, over-an-object, as user)`

File: scripts/manage_team_roles_for_user.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        description='Script to manage roles for user on per-team basis')
+    parser.add_argument('-t', '--team-name', required=True, type=str,
+                        help='Team name')
+    parser.add_argument('-e', '--email', required=True, type=str,
+                        help='User e-mail')
+    parser.add_argument('-a', '--add-role', required=False, action='append',
+                        dest='add_roles', type=str, help='Add role(s)',
+                        choices=role_names)
+    parser.add_argument('-r', '--remove-role', required=False, action='append',
+                        dest='remove_roles', type=str, help='Add role(s)',
```
Comment (javihernandez) [inline scripts/manage_team_roles_for_user.py:31]: help should be 'Remove role(s)'

**Review by soksanichenko — APPROVED** [review]
lgtm if Javier's comments will be fixed

File: scripts/manage_team_roles_for_user.py (hunk 2 above)
Comment (Korulag) [reply scripts/manage_team_roles_for_user.py:78]: Yep :)

File: alws/app.py (hunk 3 above)
Comment (Korulag) [reply alws/app.py:37]: This is automatically applied, like added to application

File: alws/crud/build.py (hunk 4 above)
Comment (Korulag) [reply alws/crud/build.py:44]: The approach was inspired by casbin examples, like "object subject action"

File: scripts/manage_team_roles_for_user.py (hunk 5 above)
Comment (Korulag) [reply scripts/manage_team_roles_for_user.py]: Fixed

**Review by anfimovdm — APPROVED** [review]
lgtm

File: alws/app.py (hunk 3 above)
Comment (javihernandez) [reply alws/app.py:37]: Got it, thanks for the clarification

File: scripts/manage_users.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                    models.UserRole.name.in_(
+                        [team_crud.get_team_role_name(team.name, r)
+                         for r in arguments.remove_roles]))
+                )).scalars().all()
+
+            if add_roles:
+                user.roles.extend(add_roles)
+            if remove_roles:
+                roles_ids = [r.id for r in remove_roles]
+                await db.execute(delete(models.UserRoleMapping).where(
```
Comment (anfimovdm) [inline scripts/manage_users.py:108]: can we delete user roles through `user.roles.remove(role)`?

**Review by anfimovdm — CHANGES_REQUESTED** [review]

File: scripts/manage_users.py (hunk 6 above)
Comment (Korulag) [reply scripts/manage_users.py:108]: Unfortunately it doesn't work that way because of many-to-many relationship

**Review by anfimovdm — APPROVED** [review]

**Review by soksanichenko — APPROVED** [review]
lgtm, because all comments are resolved

**Pull request merged**